src/main/java/Deadline.java

```diff
@@ -0,0 +1,12 @@
+public class Deadline extends Task {
+    protected String by;
+
+    public Deadline(String description, String by) {
+        super(description);
+        this.by = by;
+    }
+
+    public String toString() {
+        return "[D][" + super.getStatusIcon() + "] " + super.toString() + " (by: " + by + ")";
```

Comment:
maybe you should change "[D][" to a constant in order to reduce the need of magic strings.